doc/develop/application/index.rst

```diff
@@ -67,107 +67,104 @@ generated in a build directory; Zephyr does not support "in-tree" builds.
 The following sections describe how to create, build, and run Zephyr
 applications, followed by more detailed reference material.
 
+.. _zephyr-app-types:
 
-.. _source_tree_v2:
+Application types
+*****************
 
-Source Tree Structure
-*********************
-
-Understanding the Zephyr source tree can be helpful in locating the code
-associated with a particular Zephyr feature.
-
-At the top of the tree there are several files that are of importance:
-
-:file:`CMakeLists.txt`
-    The top-level file for the CMake build system, containing a lot of the
-    logic required to build Zephyr.
-
-:file:`Kconfig`
-    The top-level Kconfig file, which refers to the file :file:`Kconfig.zephyr`
-    also found at the top-level directory.
-
-    See :ref:`the Kconfig section of the manual <kconfig>` for detailed Kconfig
-    documentation.
-
-:file:`west.yml`
-    The :ref:`west` manifest, listing the external repositories managed by
-    the west command-line tool.
-
-The Zephyr source tree also contains the following top-level
-directories, each of which may have one or more additional levels of
-subdirectories which are not described here.
+Based on where the source code of the application is located we can distinguish
+between three basic application types.
 
-:file:`arch`
-    Architecture-specific kernel and system-on-chip (SoC) code.
-    Each supported architecture (for example, x86 and ARM)
-    has its own subdirectory,
-    which contains additional subdirectories for the following areas:
+You can find out more about how the build system supports all the application
+types described in this section in the :ref:`cmake_pkg` section.
 
-    * architecture-specific kernel source files
-    * architecture-specific kernel include files for private APIs
+.. _zephyr-repo-app:
 
-:file:`soc`
-    SoC related code and configuration files.
-
-:file:`boards`
-    Board related code and configuration files.
-
-:file:`doc`
-    Zephyr technical documentation source files and tools used to
-    generate the https://docs.zephyrproject.org web content.
+Zephyr repository application
+=============================
 
-:file:`drivers`
-    Device driver code.
+The default Zephyr :ref:`west workspace <west-workspaces>` has the following
```

Comment:
See next comment

```suggestion
In case of a Zephyr repository application the :ref:`west workspace <west-workspaces>` has the following
```

Reply:
I'm not sure which additional information we are trying to add, implementing this suggestion, as this edit can't really be done without changing also lines 104-108. 
Just check this block:

```
The default Zephyr :ref:`west workspace <west-workspaces>` has the following
structure:
...
Any application located within the ``zephyr`` folder is referred to as a Zephyr
repository application.
For example, the ``hello_world`` :ref:`sample <hello_world>` is a Zephyr
repository application.
```
Compared to this block (not making much sense):

```
In case of a Zephyr repository application the :ref:`west workspace <west-workspaces>` has the following
structure:
...
Any application located within the ``zephyr`` folder is referred to as a Zephyr
repository application.
For example, the ``hello_world`` :ref:`sample <hello_world>` is a Zephyr
repository application.
```